src/librustdoc/test.rs

```diff
@@ -247,10 +248,18 @@ fn runtest(test: &str, cratename: &str, cfgs: Vec<String>, libs: SearchPaths,
     cfg.extend(config::parse_cfgspecs(cfgs));
     let libdir = sess.target_filesearch(PathKind::All).get_lib_path();
     let mut control = driver::CompileController::basic();
+
     if no_run {
         control.after_analysis.stop = Compilation::Stop;
     }
-    driver::compile_input(sess, &cstore, cfg, &input, &out, &None, None, control);
+
+    match thread::spawn(move || {
+        driver::compile_input(sess, &cstore, cfg, &input, &out, &None, None, control);
+    }).join() {
+        Ok(_) if compile_fail           => panic!("couldn't compile the test"),
+        Err(_) if compile_fail == false => panic!("test compiled while it wasn't supposed to"),
```

Comment:
Doesn't Ok here mean that the code does compile (insofar as compilation doesn't panic)? The match here seems to indicate the opposite.